### PR TITLE
Add Replication Reindexing Progress Bar

### PR DESCRIPTION
**Pull request by noelledaley**
# Add Reindexing Progress Bar
This PR adds a progress bar to the DR & Performance Primary dashboards and the Performance Secondary dashboard when the current cluster is reindexing. The PR also includes some cleanup:

1. I made small changes so progress bars are more consistent across vault, including in the shamir flow process and in the UI wizard.
1. Moved the margin styles out of AlertBanners and into the ReplicationDashboard component so the margins are applied to every instance of the dashboard
1. Made the headers on the primary and secondary cards the same size
🧹 ✨ 😂 

I looked into building a new progress bar component but since our needs are so basic I opted to use the built in html5 progress bar (yay, semantic html!). In the future if we want to add animations we could pull this into a new component and reconstruct it using divs. A fun sideproject for another time. 

### In the UI
image

### In Storybook
image

### CODE REVIEW

File: ui/README.md [hunk 1]
```diff
@@ -158,7 +159,7 @@ Note that placing a param inside brackets (e.g. `[closedLabel=More options]` ind
 
 2. Generate a new story with `ember generate story [name-of-component]`
 3. Inside the newly generated `stories` file, add at least one example of the component. If the component should be interactive, enable the [Storybook Knobs addon](https://github.com/storybooks/storybook/tree/master/addons/knobs).
-4. Generate the `notes` file for the component with `yarn gen-story-md [name-of-component]` (e.g. `yarn gen-md alert-banner`). This will generate markdown documentation of the component and place it at `vault/ui/stories/[name-of-component].md`. If your component is a template-only component, you will need to manually create the markdown file.
+4. Generate the `notes` file for the component with `yarn gen-story-md [name-of-component] [name-of-engine-or-addon]` (e.g. `yarn gen-md alert-banner core`). This will generate markdown documentation of the component and place it at `vault/ui/stories/[name-of-component].md`. If your component is a template-only component, you will need to manually create the markdown file.
```
Comment (Monkeychip) [inline ui/README.md:162]: 🎉 Thank you for adding this.

File: ui/app/styles/components/replication-dashboard.scss [hunk 2]
```diff
@@ -15,7 +15,13 @@
     margin-bottom: $spacing-xs;
   }
 
+  .reindexing-alert,
+  .syncing-alert {
+    margin-top: 2rem;
```
Comment (Monkeychip) [inline ui/app/styles/components/replication-dashboard.scss:20]: I'm totally fine with this being 2rem, but would anything in the helpers.scss file work, e.g. 
```
.has-top-margin-xl {
  margin-top: $spacing-xl;
}
```

Maybe not use the class has-top-margin-xl, but replace the 2rem with $spacing-xl?

File: ui/lib/core/addon/templates/components/replication-dashboard.hbs [hunk 3]
```diff
@@ -1,14 +1,16 @@
...
+        @title={{concat "Re-indexing in progress" reindexingStage}}
+        @type="info"
+        @progressBar={{progressBar}}
+        @message="This can cause a delay depending on the size of the data store. You can use Vault during this time."
+        data-test-isReindexing />
+    </div>
   {{/if}}
 
-  <div class="selectable-card-container has-top-margin-xl">
+  <div class="selectable-card-container">
```
Comment (Monkeychip) [inline ui/lib/core/addon/templates/components/replication-dashboard.hbs:13]: Curious where the margin is now coming from for this view: 

Screen Recording 2020-05-12 at 11 12 AM

File: ui/tests/integration/components/replication-dashboard-test.js [hunk 4]
```diff
@@ -25,34 +33,50 @@ module('Integration | Enterprise | Component | replication-dashboard', function(
...
+    await render(hbs`<ReplicationDashboard
+      @replicationDetails={{replicationDetails}}
+      @clusterMode={{clusterMode}}
+      @isSecondary={{isSecondary}}
+      @componentToRender='replication-secondary-card'
     />`);
 
     assert.dom('[data-test-isSyncing]').exists();
-    assert.dom('[data-test-isReindexing]').doesNotExist();
+    assert
```
Comment (Monkeychip) [inline ui/tests/integration/components/replication-dashboard-test.js:58]: nice.  I had also written a test for reindexing, so I will remove it.

File: ui/app/styles/components/replication-dashboard.scss (hunk 2 above)
Comment (noelledaley) [reply ui/app/styles/components/replication-dashboard.scss:20]: good call, i'll use $spacing-xl. thank you!

File: ui/app/styles/components/replication-dashboard.scss [hunk 5]
```diff
@@ -15,7 +15,13 @@
...
+  }
+
   .selectable-card-container {
+    margin-top: 2rem;
```
Comment (noelledaley) [inline ui/app/styles/components/replication-dashboard.scss:24]: i'll update this one too

File: ui/lib/core/addon/templates/components/replication-dashboard.hbs (hunk 3 above)
Comment (noelledaley) [reply ui/lib/core/addon/templates/components/replication-dashboard.hbs:13]: excellent question! i removed the `has-top-margin-xl` classes from the AlertBanners and instead applied them to the dashboard component itself. this way all instances of the dashboard have consistent margins and paddings. that change is right here: https://github.com/hashicorp/vault/pull/8975/files#diff-ad14bf8f91b12ee633f19fcf48abf7caR24

before that change the primary & dr secondary dashboards had slightly different top margins, and i thought it was better to keep the style tightly coupled to the dashboard component itself vs. manually applying styles to other components.

**Pull request merged**